### DISCUSSION TITLE
Data flow Analysis in Go

### DISCUSSION THREAD

**Discussion by 0xTeles** [github/securitylab#684] [Q&A] [answered]
Hello folks, how are you doing? I'm begineer on CodeQL (I start learning in past week) and I trying solve the CTF "Go and Don't Return" but I can't understand the Data Flow Analysis in Go. I saw the answer of first question of second part but I don't understand. You have or do you know any blog post to help me understand this? Thanks!!!

**Comment by edoardopirovano** [comment, marked as answer]
Greetings! Glad to hear you're learning CodeQL - the learning curve at the start can be a little bit steep but it's definitely worth it! In terms of resources, I assume you've already seen [these tutorials](https://codeql.github.com/docs/writing-codeql-queries/ql-tutorials/#ql-tutorials) - they're not specific to dataflow analysis but are definitely worth completing those before you move on to more complex things like the CTF you are currently working on.

There's some information on dataflow in Go [here](https://codeql.github.com/docs/codeql-language-guides/modeling-data-flow-in-go-libraries/). Also, the dataflow libraries are pretty similar across different languages so the [other dataflow tutorials](https://codeql.github.com/docs/writing-codeql-queries/about-data-flow-analysis/) are also likely to give you helpful background that can be transferred to Go.

Finally, there's also an excellent blog post that @jorgectf wrote with some more resources on learning CodeQL [here](https://jorgectf.github.io/blog/post/practical-codeql-introduction/): again, it's not specific to Go but you'll find that most of what you learn will be transferable.

**Reply by 0xTeles** [reply]
Thanks so much for the guide, @edoardopirovano. I will read all references and keep learning about CodeQL.
